bolt-modules/boltlib/lib/puppet/functions/set_var.rb

```diff
@@ -0,0 +1,30 @@
+# Sets a variable { key => value } for a target
+#
+# This function takes 3 parameters:
+# * A Target object to set the variable for
+# * The key for the variable (String)
+# * The value of the variable (Data)
+#
+# Returns nil
```

Comment:
From the perspective of the puppet language the return is `undef` not `nil` (since `nil` becomes `undef` in puppet).